contrib/vagrant/scripts/helpers.bash

```diff
@@ -103,7 +103,8 @@ function download_to {
         log "Downloading ${component}..."
 
         rm -f "/tmp/${component}"
-        ${WGET} -O "/tmp/${component}" -nv "${url}"
+        # fix: No such file or directory
+        eval ${WGET} -O "/tmp/${component}" -nv "${url}"
         # Hide 'failed to preserve ownership' error
         mv "/tmp/${component}" "${cache_dir}/${component}" 2>/dev/null
```

Comment:
This code looks like it is intended to download a component and place it in a certain directory. I don't understand how eval makes a difference. Can you describe the intent here?

Reply:
The "eval" command will first scan the command line for all substitutions, and then execute the command. fixed "No such file or directory" error when configuring "vm_set_proxy”

You can simply reproduce this error problem by executing the following command to simulate the error
```shell
cat >> test.sh << 'EOF'
export WGET="HTTPS_PROXY=http://you.proxy.domain wget"
${WGET} https://www.google.com
EOF
chmod +x test.sh
./test.sh
```

Reply:
Is it possible to solve this without `eval`?

I was wondering perhaps whether it's sufficient to add a line something like this to the top of the file:

```
HTTPS_PROXY={HTTPS_PROXY:-""}
```

Or, similar to the other scripts, change the WGET commandline based on the VM_PROXY variable?

Reply:
Currently testing adding eval here can solve this problem with minimal changes to the code

Reply:
^^ Mostly I'm just looking at the original PR https://github.com/cilium/cilium/pull/7289 and wondering why the `eval` is necessary. This seems like the code is intended for the caller to pass in a `WGET` variable so that we don't have to `eval` here. Maybe @jomenxiao could chime in on how they made this work before and what the best solution is?